addons/account/models/account_bank_statement.py

```diff
@@ -16,7 +16,7 @@ def default_get(self, fields_list):
         # create statement on a saved statement line in the tree view
         if self._context.get('st_line_id'):
             st_line = self.env['account.bank.statement.line'].browse(self._context['st_line_id'])
-            defaults['balance_start'] = st_line.running_balance - st_line.amount
+            defaults['balance_start'] = st_line.running_balance - (st_line.amount if st_line.state == 'posted' else 0)
```

Comment:
Can you create a bank statement from a cancelled/draft statement line?

Reply:
If it is decided to display the draft/canceled lines in the future, this will cause a bug which won't be easy to notice. So it is at least a preventive check. Also, there could be possible to that by canceling a move in one tab and making an statement from it in another tab.